### PR TITLE
[tests] More testing

### PR DESCRIPTION
**Pull request by Scott8440**
## Summary
More unit tests

## Test Plan
the unit tests

### CODE REVIEW

File: alertmanager/config/config.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func (c *Config) RemoveReceiverFromRoute(receiver string) {
 	for i, route := range c.Route.Routes {
 		c.Route.Routes[i] = removeReceiverFromRouteImpl(receiver, route)
 	}
-	prunedRoutes := funk.Filter(c.Route.Routes, func(x *Route) bool {return x != nil})
+	prunedRoutes := funk.Filter(c.Route.Routes, func(x *Route) bool { return x != nil })
```
Comment (rckclmbr) [inline alertmanager/config/config.go:116]: This should be linted / auto formatted.  If you committed it incorrectly, should a test have failed?

File: alertmanager/config/config_test.go [hunk 2]
```diff
@@ -8,17 +8,19 @@
...
 	"github.com/stretchr/testify/assert"
 )
 
 var (
 	testConfig = Config{
-		Global:       nil,
-		Route:        &Route{
-			Receiver:       "base",
-			Routes:         []*Route{
+		Global: nil,
```
Comment (rckclmbr) [inline alertmanager/config/config_test.go:20]: Same

**Review by rckclmbr — APPROVED** [review]

File: alertmanager/config/config.go (hunk 1 above)
Comment (Scott8440) [reply alertmanager/config/config.go:116]: I guess this isn't a lint that is actually checked by gofmt or the linter. Personal preference?

File: alertmanager/config/config_test.go (hunk 2 above)
Comment (Scott8440) [reply alertmanager/config/config_test.go:20]: This probably changed formatting because a lot of stuff was added to it, changing the spacing

**Pull request merged**